### PR TITLE
Docs and tweaks to Angular deploy

### PR DESCRIPTION
**Pull request by MocanaAtBroad**
Scope has broadened on this PR.
This the main PR that contains changes to build deploy of Angular app on GAE.

### CODE REVIEW

File: configure.rb [hunk 1]
```diff
@@ -181,25 +181,25 @@ def render_from_path(path, output_file_name = nil)
...
+        $dsde_toolbox_image_name,
+        "consul-template", "-config=/etc/consul-template/config/config.json", "-vault-retry-attempts=3",
+        "-template=#{file_name}:#{output_file_name}",
+        "-once"
     ]
   else
-    vault_cmd = ["consul-template", "-config=/etc/consul-template/config/config.json",
-                              "-template=#{file_name}:#{output_file_name}", "-once"]
+    vault_cmd = ["consul-template", "-config=/etc/consul-template/config/config.json", "-vault-retry-attempts=3",
+                 "-template=#{file_name}:#{output_file_name}", "-once"]
```
Comment (MocanaAtBroad) [inline configure.rb:202]: Vault token had expired and consul-template was retrying forever to retrieve vault info. This sets limit.

File: admin/DDPAngularAppDeployerRole.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+- cloudbuild.builds.create
+- cloudbuild.builds.get
+- cloudbuild.builds.list
+- cloudbuild.builds.update
+- storage.objects.create
+- storage.objects.delete
+- storage.objects.get
+- storage.objects.list
+stage: GA
+title: DDP Angular App Deployer
```
Comment (zyme) [inline admin/DDPAngularAppDeployerRole.yaml]:
```suggestion
title: DDP Angular CI CD
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -51,6 +50,9 @@ commands:
               develop)
                 DEPLOY_ENV=dev
               ;;
+              angular-appengine-deployment-ctd-2)
+                DEPLOY_ENV=dev
+              ;;
```
Comment (MocanaAtBroad) [inline .circleci/config.yml:55]: For testing purposes. Will go away eventually

File: .circleci/config.yml [hunk 4]
```diff
@@ -87,7 +89,7 @@ commands:
           command: |
             set -u
             # we are getting our builds from the one bucket
-            readvault.sh secret/pepper/${ENVIRONMENT}/v1/conf .data.gcp.serviceKey | gcloud auth activate-service-account --key-file=-
+            readvault.sh secret/pepper/prod/v1/conf .data.gcp.serviceKey | gcloud auth activate-service-account --key-file=-
             TAR_FILE_URL_PATTERN="gs://${BUILDS_BUCKET}/${ANGULAR_PROJECT_NAME}/${ANGULAR_PROJECT_NAME}_*_${SHORT_GIT_SHA}.tar.gz"
```
Comment (MocanaAtBroad) [inline .circleci/config.yml:93]: It's always the prod key when getting things in and out of build bucket.

File: .circleci/config.yml [hunk 5]
```diff
@@ -166,6 +168,7 @@ commands:
       - run:
           name: Set environment variables to build <<parameters.study_key>>
           command: |
+            echo "export VAULT_TOKEN=`vault write -field=token auth/approle/login role_id=$VAULT_ROLE_ID secret_id=$VAULT_ROLE_SECRET_ID`" >> $BASH_ENV
             echo 'export PATH=$PATH:~/repo/build-utils' >> $BASH_ENV
```
Comment (MocanaAtBroad) [inline .circleci/config.yml:172]: We now generate the vault token using Vault approle  . role_id and secret_id come from project settings.

File: admin/DDPCICDRole.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+- appengine.versions.create
+- appengine.versions.delete
+- appengine.versions.get
+- appengine.versions.list
+- appengine.versions.update
+- cloudbuild.builds.create
+- cloudbuild.builds.get
+- cloudbuild.builds.list
+- cloudbuild.builds.update
+- storage.objects.create
```
Comment (zyme) [inline admin/DDPCICDRole.yaml:20]: Can we remove the `storage.*` items here?  We ended up taking care of this with specific permissions for the prod bucket.

**Review by zyme — DISMISSED** [review]

File: admin/DDPCICDRole.yaml (hunk 6 above)
Comment (MocanaAtBroad) [reply admin/DDPCICDRole.yaml:20]: I'll have to test it out. I think we needed some of those just for deployment.

Comment (MocanaAtBroad) [reply admin/DDPCICDRole.yaml:20]: We need them!

**Review by zyme — APPROVED** [review]

**Pull request merged**